tests/test_qnode.py

```diff
@@ -1482,6 +1483,57 @@ def qn2(x, y):
         assert tape.shots.total_shots == total_shots
         assert tape.shots.shot_vector == shot_vector
 
+    def test_shots_initialization(self):
+        """Test that _shots is correctly initialized from the device."""
+        dev = qml.device("default.qubit", wires=1, shots=42)
+        qn = qml.QNode(dummyfunc, dev)
+        assert qn._shots == qml.measurements.Shots(42)
+
+        dev_analytic = qml.device("default.qubit", wires=1)
+        qnode_analytic = qml.QNode(dummyfunc, dev_analytic)
+        assert qnode_analytic._shots is None or qnode_analytic._shots.total_shots is None
+
+    def test_shots_update_with_device(self):
+        """Test that _shots is updated when updating the QNode with a new device."""
+        dev1 = qml.device("default.qubit", wires=1, shots=100)
+        qn = qml.QNode(dummyfunc, dev1)
+        assert qn._shots == qml.measurements.Shots(100)
+
+        dev2 = qml.device("default.qubit", wires=1, shots=200)
+        updated_qnode = qn.update(device=dev2)
+        assert updated_qnode._shots == qml.measurements.Shots(200)
```

Comment:
This is actually kinda a weird combination of effects, and I'm actually 100% certain what behavior I would expect here.